### ISSUE TITLE
`create_ticket` does not support binary attachments

### ISSUE THREAD

**Issue by haiiiiiyun** [sjkingo/python-freshdesk#36]
Only support plain text files, does not support binary files, such as image files.

**Comment by sjkingo** [comment]
Hi, are you able to provide a simple reproducible case and the complete output to show the error you're receiving?
Thanks

**Comment by haiiiiiyun** [comment]
```python
import os

from freshdesk.api import API  # version 1.2.3
a = API('mydomain.freshdesk.com', 'key', version=2)

path = os.path.abspath(os.path.dirname(__file__))

ticket = a.tickets.create_ticket('Translation request',
                             email='example@example.com',
                             description='I need this article be translated',
                             tags=["translation", "request"],
                             attachments=[
                                     os.path.join(path, "attach1.txt"),
                                     os.path.join(path, "attach.png"),
                             ],
                         )
```

This is the example script, `attach1.txt` is a file with unicode characters, got error `UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte`

**Comment by sjkingo** [comment]
Thanks. Could you please provide the result of running your commit against both master and 1.2.3?

**Comment by haiiiiiyun** [comment]
The only change is in file `freshdesk/v2/api.py`,  in the `_create_ticket_with_attachment()` function, see below:

```python
def _create_ticket_with_attachment(self, url, data):
        attachments = data['attachments']
        del data['attachments']
        multipart_data = []

        # support tags
        for k, v in [(i,j) for i,j in data.items()]:
            if type(v) == list:
                data[k+"[]"] = v
                data.pop(k)

        for attachment in attachments:
            file_name = attachment.split("/")[-1:][0]
            multipart_data.append(('attachments[]', (file_name, open(attachment, 'rb'), None))) # support binary attachment


        ticket = self._api._post(url, data=data, files=multipart_data)
        return ticket
```

**Comment by ArtemGordinsky** [comment]
This issue could be closed since @sjkingo released 1.2.4
